### DISCUSSION TITLE
How can I do something when animation is completed?

### DISCUSSION THREAD

**Discussion by kyuranger** [AvaloniaUI/Avalonia#12892] [Q&A]
Here is my code:
```
<Border Background="Black" Opacity="1" Name="BackgroundB">
				<Border.Transitions>
					<Transitions>
						<DoubleTransition Property="Opacity" Duration="0:0:0.3"></DoubleTransition>
					</Transitions>
				</Border.Transitions>
</Border>
```

```
public void Hide()
{
    BackgroundB.Opacity=0;
}
```

When the void `Hide()` works, I want to set the `IsVisiable` of the `BackgroundB` to false when its OpacityAnimation is complete.

How can I achieve this?

**Comment by kyuranger** [comment]
I found a stupid solution:
```
Task.Run(()=>{
    Task.Delay(300),Wait();
    Dispatcher.UIThread.Invoke(()=>{
    //Do Something.
});
});
```

**Reply by timunie** [reply]
Better not use `Task.Delay`. And even worth, don't use `.Wait()` 

I suggest to use `DispatcherTimer` instead. on `Tick`´-event, do your stuff and stop timer immediately.

**Reply by kyuranger** [reply]
Could you please tell me why not to use it? I hope to understand the detailed reasons and learn more.

**Reply by timunie** [reply]
Wait can cause a deadlock, see: https://blog.stephencleary.com/2012/07/dont-block-on-async-code.html

**Reply by kyuranger** [reply]
Thank you.
